### PR TITLE
Maven test data generator

### PR DESCRIPTION
**Pull request by ChGen**
Here is the maven test data generator as new script: `prep-maven-dyn.py`. Script description is added in `README.md`. 
Also, I'm updating the scripts/directories structure, and it will be provided as the separate pull request.
This is part of https://github.com/artipie/benchmarks/issues/59

### CODE REVIEW

**Comment by ChGen** [comment]
Example of generated directory structure:
```
tree test-data/maven-dyn/repository

test-data/maven-dyn/repository
├── group1
│   ├── artifact1-big
│   │   ├── 1.1.0
│   │   │   ├── artifact1-big-1.1.0.jar
│   │   │   ├── artifact1-big-1.1.0.jar.md5
│   │   │   ├── artifact1-big-1.1.0.jar.sha1
│   │   │   ├── artifact1-big-1.1.0.pom
│   │   │   ├── artifact1-big-1.1.0.pom.md5
│   │   │   └── artifact1-big-1.1.0.pom.sha1
│   │   ├── 1.2.0
│   │   │   ├── artifact1-big-1.2.0.jar
│   │   │   ├── artifact1-big-1.2.0.jar.md5
│   │   │   ├── artifact1-big-1.2.0.jar.sha1
│   │   │   ├── artifact1-big-1.2.0.pom
│   │   │   ├── artifact1-big-1.2.0.pom.md5
│   │   │   └── artifact1-big-1.2.0.pom.sha1
│   │   ├── 1.3.0
│   │   │   ├── artifact1-big-1.3.0.jar
│   │   │   ├── artifact1-big-1.3.0.jar.md5
│   │   │   ├── artifact1-big-1.3.0.jar.sha1
│   │   │   ├── artifact1-big-1.3.0.pom
│   │   │   ├── artifact1-big-1.3.0.pom.md5
│   │   │   └── artifact1-big-1.3.0.pom.sha1
│   │   ├── maven-metadata.xml
│   │   ├── maven-metadata.xml.md5
│   │   └── maven-metadata.xml.sha1
│   ├── artifact1-med
│   ├── artifact2-med
│   ├── artifact3-med
│   └── artifact4-med
├── group2
│   ├── artifact1-med
│   ├── artifact2-med
│   ├── artifact3-med
│   └── artifact4-med
└── group3
    ├── artifact1-med
    ├── artifact1-small
    ├── artifact2-small
    ├── artifact3-small
    └── artifact4-small
```

**Comment by ChGen** [comment]
@dgarus , please review. Here's the script itself with some basic descriptions.

**Comment by ChGen** [comment]
@olenagerasimova , please review

**Review by olenagerasimova — DISMISSED** [review]
@ChGen thanks, I'm not expert in Python, looks good. Though I'm not sure that we actually need so many parameters - amount of groups, artifacts and versions per group...

**Comment by ChGen** [comment]
@Swizbiz , @dgarus  do you have any extra notes on this PR?

File: loadtests/README.md [hunk 1]
```diff
@@ -9,12 +9,48 @@ For now using the following version of JMeter:
...
+
+Maven test data generator `prep-maven-dyn.py` requires all generator parameters provided for the command line. Here's the command format:
+```
+./gen-maven.py groups artifacts versions big_size med_size small_size bigs_percent smalls_percent
+```
+For example:
+```
+./prep-maven-dyn.py 4 5 6 9000000 192000 16386 5 30
+```
+Paramets description:
```
Comment (dgarus) [inline loadtests/README.md]: Are these all parameters mandatory?

File: loadtests/README.md [hunk 2]
```diff
@@ -9,12 +9,48 @@ For now using the following version of JMeter:
...
+- `groups` number of maven groups to generate 
+- `artifacts` number of artifacts inside the group to generate
+- `versions` number of versions per artifactId
+- `big_size` size of "big" artifact type in bytes
+- `med_size` size of "medium" artifact type in bytes
+- `small_size` size of "small" artifact type in bytes
+- `bigs_percent` percent of "big" artifacts (jars) to generate
+- `smalls_percent` percent of "small" artifacts. The rest will be "medium" sized.
```
Comment (dgarus) [inline loadtests/README.md]: `mediums_percen` looks more logical for me

File: loadtests/README.md (hunk 1 above)
Comment (ChGen) [reply loadtests/README.md]: @dgarus , yes, all parameters are required.

Comment (dgarus) [reply loadtests/README.md]: > @dgarus , yes, all parameters are required.

I think we should have reasonable default values.

For example, `prep-maven-dyn.py --artifact_numbers 10_000`
and all other parameters should be calculated or have default values.
But the option to define all parameters should be too.

File: loadtests/README.md (hunk 2 above)
Comment (ChGen) [reply loadtests/README.md]: @dgarus , ok, I'll switch to specifying bigs & mediums, so small ones will be calculated.

File: loadtests/prep-maven-dyn.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    BIG = 'big'
+    MED = 'med'
+    SMALL = 'small'
+
+    totals = int(groups * artifacts * versions)
+    bigs = int(totals * bigP / 100)
+    smalls = int(totals * smallP / 100)
+    mediums = int(totals - bigs - smalls)
+    if totals <= 0 or mediums < 0:
+        print(f"Error. Incorrent artifacts counts: {totals}; {bigs}; {mediums}; {smalls}", file=sys.stderr)
```
Comment (dgarus) [inline loadtests/prep-maven-dyn.py]: Printing a help description would be fine

Comment (ChGen) [reply loadtests/prep-maven-dyn.py]: ok

File: loadtests/README.md (hunk 1 above)
Comment (ChGen) [reply loadtests/README.md]: @dgarus , ok, I'll think on flexible settings

File: loadtests/prep-maven-dyn.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        exit(1)
+
+    print(f"Limits: {totals}; {bigs}; {smalls}, {mediums}")
```
Comment (Swizbiz) [inline loadtests/prep-maven-dyn.py]: Add info for every digits. Like a big=20, medium=100 and etc.
And order should be big, medium, small.

File: loadtests/README.md [hunk 5]
```diff
@@ -9,12 +9,48 @@ For now using the following version of JMeter:
...
-time mvn clean install -Dhttp.proxyHost=localhost -Dhttp.proxyPort=8888 -Dhttps.proxyHost=localhost -Dhttps.proxyPort=8888 -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.skip -Ddockerfile.skip=true
+## Maven test data generator parameters
+
+Maven test data generator `prep-maven-dyn.py` requires all generator parameters provided for the command line. Here's the command format:
+```
+./gen-maven.py groups artifacts versions big_size med_size small_size bigs_percent smalls_percent
+```
+For example:
+```
+./prep-maven-dyn.py 4 5 6 9000000 192000 16386 5 30
```
Comment (Swizbiz) [inline loadtests/README.md]: I think we should add a name for the parameters. Because right now it is not clear what is what in all the parameters.

**Review by Swizbiz — COMMENTED** [review]
@ChGen thank you.
Please, look at my comments.

File: loadtests/README.md (hunk 5 above)
Comment (ChGen) [reply loadtests/README.md]: @Swizbiz , thanks, will switch to named parameters.

File: loadtests/README.md [hunk 6]
```diff
@@ -9,12 +9,50 @@ For now using the following version of JMeter:
...
+```
+./prep-maven-dyn.py --total-artifacts 100
+./prep-maven-dyn.py --groups 3 --artifacts 4 --versions=5
+```
 
+All available generator parameters (available by `--help` key):
+```
+  --total-artifacts TOTAL_ARTIFACTS  Specify tatal artifacts count. May override groups/artifacts/versions count. (default: None)
+  --groups GROUPS           Count of maven groups to generate (default: 3)
+  --artifacts ARTIFACTS
```
Comment (Swizbiz) [inline loadtests/README.md]: image

Delete extra new line at Artifacts line

Comment (ChGen) [reply loadtests/README.md]: Oh, my bad :(

**Comment by ChGen** [comment]
@dgarus , @Swizbiz , I switched to named arguments with defaults. Please, review.

**Review by Swizbiz — APPROVED** [review]
Thank you!

**Review by dgarus — APPROVED** [review]
thanks! lgtm

**Pull request merged**